webapp/src/pages/Settings.tsx

```diff
@@ -31,29 +62,540 @@ const Settings: React.FC = () => {
+              [metricName]: {
+                class_name: "datasets.load_metric",
+                kwargs: {
+                  path: metricName.toLowerCase(),
+                },
+                additional_kwargs: ADDITIONAL_KWARGS_CUSTOM_METRICS.includes(
+                  metricName
+                )
+                  ? { average: "weighted" }
+                  : {},
+              },
+            }),
+          },
+        })
+      : setPartialConfig({
+          ...partialConfig,
+          ...{
+            metrics: _.omit(resultingConfig.metrics, metricName),
+          },
+        });
+  };
+
+  const getProjectConfigSection = () => (
+    <Box display="flex" flexDirection="column">
+      {displaySectionTitle("General")}
+      <Box
+        display="flex"
+        flexDirection="row"
+        justifyContent="flex-start"
+        marginLeft={2}
+      >
+        {displayReadonlyFields("name", resultingConfig.name)}
+        {displayReadonlyFields(
+          "rejection class",
+          resultingConfig.rejection_class
+        )}
+      </Box>
+      <Box display="flex" flexDirection="row" marginLeft={2}>
+        {displaySubSectionTitle("Columns")}
+        {resultingConfig.columns &&
+          Object.entries(resultingConfig.columns).map(
+            ([field, value], index) => (
+              <Box key={index} gap={(theme) => theme.spacing(1)}>
+                {displayReadonlyFields(field, value)}
+              </Box>
+            )
+          )}
+      </Box>
+      {divider}
+      {displaySectionTitle("Dataset")}
+      <Box
+        display="flex"
+        flexDirection="row"
+        justifyContent="flex-start"
+        marginLeft={2}
+      >
+        {displayReadonlyFields(
+          "class name",
+          resultingConfig.dataset?.class_name
+        )}
+        {displayReadonlyFields("remote", resultingConfig.dataset?.remote)}
+      </Box>
+      {resultingConfig.dataset?.kwargs && (
+        <Box display="flex" flexDirection="row" marginLeft={2} gap={5}>
+          {displaySubSectionTitle("Kwargs")}
+          {Object.entries(resultingConfig.dataset?.kwargs).map(
+            ([field, value], index) => (
+              <Box key={index} gap={(theme) => theme.spacing(1)}>
+                {displayReadonlyFields(field, String(value))}
+              </Box>
+            )
+          )}
+        </Box>
+      )}
+      {resultingConfig.dataset?.args &&
+        resultingConfig.dataset?.args.length > 0 && (
+          <Box display="flex" flexDirection="row" marginLeft={2} gap={5}>
+            {displaySubSectionTitle("args")}
+            {Object.entries(resultingConfig.dataset?.args).map(
+              ([field, value], index) => (
+                <Box key={index} gap={(theme) => theme.spacing(1)}>
+                  {displayReadonlyFields(field, String(value))}
+                </Box>
+              )
+            )}
+          </Box>
+        )}
+    </Box>
+  );
+  const getModalContractConfigSection = () => (
```

Comment:
I think you mean "model", and not "modal".
```suggestion
  const getModelContractConfigSection = () => (
```